src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Subscriber.kt

```diff
@@ -42,25 +43,45 @@ internal class MessageReceiverExample : MessageReceiver {
     }
 }
 
+class SubscribeManager(subscriptionId: String) {
+    private val dockerProcessor = DockerProcessor()
+    private val subscriptionName = SubscriptionName.of(PROJECT_ID, subscriptionId)
+    private var subscriber: Subscriber? = null
+
+    fun subscribe(callback: (message: String, md5sum: String?) -> Unit) {
+        try {
+            subscriber = Subscriber.newBuilder(subscriptionName, MessageReceiverExample()).build()
+            subscriber?.startAsync()?.awaitRunning()
+
+            while (true) {
+                processMessage(callback)
```

Comment:
Let's remove this remnant from the demo. Pass your callback to the message receiver and call it directly when you receive a new message.

Reply:
Do we need queue of messages after that?

Reply:
Probably not. 
In the production use task processing will take a lot of time and CPU, and one subscriber node will be able process at most two tasks anyway, so it should not eagerly accept all incoming messages.

Reply:
How should I test subscriber in that way then?
I can pass callback, but I need to publish message to trigger it.

Reply:
Done